src/components/pages/reports/Reports.js

```diff
@@ -1,44 +1,110 @@
-    if(!reportsList.length) getUserReports();
+    this.templateId = this.props.match.params.templateIndex;
+    this.searchParams = this.props.location.search && qs.parse(this.props.location.search);
+    if (this.searchParams !== undefined){
+      this.props.setTemplateSearchParams(this.searchParams);
+    }
+    if (this.templateId) {
+      this.props.setSelectedTemplateIndex(this.templateId);
+    }
+  }
+  componentDidMount() {
+    const reportIds = this.props.templates.ids;
+    if (reportIds.length === 0) { 
+      this.props.getUserTemplates(); 
+    } else {
+      this.props.getReportAnswers(reportIds[0]);
+    }
+  }
+  componentWillReceiveProps(nextProps){
+    if (this.props.templates.ids.length !== nextProps.templates.ids.length){
+      this.props.getReportAnswers(nextProps.templates.ids[this.templateId || 0]);
+    }
+    if (this.props.match.params.templateIndex !== nextProps.match.params.templateIndex){
+      this.props.setSelectedTemplateIndex(nextProps.match.params.templateIndex);
+    }
+    if (this.props.location.search !== nextProps.location.search){
+      this.searchParams = nextProps.location.search && qs.parse(nextProps.location.search);
+      this.props.setTemplateSearchParams(this.searchParams);
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState){
+    return this.props.templates.ids.length !== nextProps.templates.ids.length || 
+           this.props.answers !== nextProps.answers || 
+           this.props.match.params.templateIndex !== nextProps.match.params.templateIndex ||
+           this.props.location.search !== nextProps.location.search;
+  }
+
+  downloadReports = () => {
+    const index = this.props.match.params.templateIndex || 0
+    this.props.downloadAnswers(this.props.templates.ids[index]);
   }
 
   render() {
-    const { reportsList } = this.props;
+    const { answers, templates } = this.props;
+    const columns = [{
+      Header: 'Lat/Long',
+      accessor: 'latLong'
+    },{
+      Header: 'Area of interest',
+      accessor: 'aoi'
+    },{  
+      Header: 'Date',
+      accessor: 'date'
+    },{
+      Header: 'Member',
+      accessor: 'member'
+    }]
+
     return (
       <div>
         <Hero
-          title="Reports"
+          title="Your reports"
+          action={{name:'Download Reports', callback: this.downloadReports}}
         />
-      <div className="l-content">
-          <Article title="Uploaded">
-            <GridGallery
-              Component={ReportCard}
-              className="report-card-item"
-              collection={reportsList}
-              columns={{ small: 12, medium: 4, large: 3 }}
-            />
-          </Article>
-        </div>
+          <div className="l-content">
+            <Article>
+              <Filters
+                templates={templates} 
+                answers={answers}
+                templateIndex={this.templateIndex}
+                searchParams={this.searchParams}
```

Comment:
I think that we should avoid passing router state as props, `withRouter` option is better, and we avoid all this logic. If we need this information on the container, we can access it from the `ownProps` param.